### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'some')

### ISSUE THREAD

**Issue by sn3h** [StaticJsCMS/static-cms#874]
**Describe the bug**
Everything works fine on localhost, but when I try to load detail of any entity in production I get UI for few ms and then this error occurs

TypeError: Cannot read properties of undefined (reading 'some')
    at Object.hasInlines (_next/static/chunks/0b54ada9-bdc2be2cffc9c36b.js:172:114725)
Opening an issue pre-populates it with the error message and debugging data.
Please verify the information is correct and remove sensitive data if exists.

Details
TypeError: Cannot read properties of undefined (reading 'some')
    at Object.hasInlines (_next/static/chunks/0b54ada9-bdc2be2cffc9c36b.js:172:114725)
    at Object.hasInlines (_next/static/chunks/0b54ada9-bdc2be2cffc9c36b.js:172:63004)
...

it points towards method

```
 hasInlines: function() {
    for (var sh, dC = arguments.length, dU = Array(dC), Tl = 0; Tl < dC; Tl++)
        dU[Tl] = arguments[Tl];
    return sh = nB,
    dU.children.some(nB=>tum.isText(nB) || tur.isInline(sh, nB))
},
```

Everything else in admin works normaly (listing, files, ...) . There is no error in network or anywhere else (even data from eny entity loads fine it seems). 

**To Reproduce**
I am not quite sure. It used to work normaly on v 2.4.3, but now it shows this in every detail of any entity. I am using GitLab pkce auth. Tried to update to latest version too, but no luck. Hosting on vercel.

**Expected behavior**
Production should work just like localhost.

**Screenshots**
![image](https://github.com/StaticJsCMS/static-cms/assets/768656/d4b87a64-c511-4ed5-87a6-d0743fd0bc23)**Applicable 
Versions:**
 - Static CMS version: `static-cms@3.2.0`
 - Git provider: `gitlab`
 - Browser version: `Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Safari/537.36`

**CMS configuration**
```
"dependencies": {
		"@babel/core": "^7.22.17",
		"@babel/preset-react": "^7.22.15",
		"@staticcms/core": "3.2.0",
		"date-fns": "^2.30.0",
		"frontmatter-markdown-loader": "^3.7.0",
		"fs-extra": "^11.1.1",
		"gray-matter": "4.0.3",
		"intersection-observer": "^0.12.2",
		"leaflet": "^1.9.4",
		"leaflet-defaulticon-compatibility": "^0.1.2",
		"next": "^13.4.19",
		"react": "^18.2.0",
		"react-anchor-link-smooth-scroll": "^1.0.12",
		"react-dom": "^18.2.0",
		"react-icons": "^4.11.0",
		"react-image-gallery": "^1.3.0",
		"react-intersection-observer": "^9.5.2",
		"react-leaflet": "^4.2.1",
		"react-markdown": "^8.0.7",
		"react-responsive-carousel": "^3.2.23",
		"rehype-raw": "^7.0.0",
		"remark": "14.0.3",
		"remark-html": "15.0.2",
		"sharp": "^0.32.5",
		"tailwindcss": "^3.3.3"
	},
	"scripts": {
		"dev": "next dev",
		"build": "next build",
		"start": "next start",
		"lint": "next lint",
		"copyimages": "node ./src/bin/copy-images.mjs",
		"prebuild": "npm run copyimages"
	},
	"devDependencies": {
		"@next/eslint-plugin-next": "^13.4.19",
		"@types/node": "20.6.0",
		"autoprefixer": "^10.4.15",
		"eslint-config-next": "^13.4.19",
		"eslint-plugin-react": "^7.33.2",
		"postcss": "^8.4.29",
		"postcss-css-variables": "^0.19.0",
		"postcss-import": "^15.1.0",
		"typescript": "5.2.2"
	}
```

**Comment by KaneFreeman** [comment]
This appears to be an issue with the markdown editor (at least that is where the stack trace is pointing to). I am not sure why, potentially some issue in the Plate/Slate dependency code. I'll update those dependencies for the next release and we can see if that resolves the issue for you.

**Comment by KaneFreeman** [comment]
Version `3.3.0` has been released with updated versions of the `plate` and `slate` dependencies. Please try that version and let me know if you are still seeing the issue or not.

**Comment by sn3h** [comment]
Hello, thanks for the fast response and support. Unfortunatelly it seems it didn't help.  I think it's some problem with GitLab x Vercel as we've tried it with Netlify identity (which we left before because Netlify was buggy with building nextjs v13) and it seems to be working.

Probably gonna switch back to Netlify Identity. I'll try to keep you posted

**Comment by michal-cic** [comment]
I'm facing the same issue as OP, even with version `3.3.0`. It indeed seems to be a problem with the markdown editor/field, because removing it from a collection makes entries load without a problem.

**Comment by KaneFreeman** [comment]
Could either of you provide the following to help me reproduce this.

1. Your config for the collection having issues
2. A sample data file from that collection

**Comment by michal-cic** [comment]
Here's a [reproduction repo](https://github.com/michal-cic/staticcms-md-bug-reproduction)

**Comment by KaneFreeman** [comment]
Thanks! I was able to reproduce it and will work to find a fix.

Also, I did notice that in dev the CMS get loaded twice due to `reactStrictMode` being on (it renders everything twice on purpose to check for side effects). See https://github.com/vercel/next.js/issues/35822

If you want to keep `reactStrictMode` on you need to check to see if the CMS is already loaded in the `CMSView` before initializing. You can do that by checking for the id `cms-root`,

**Comment by KaneFreeman** [comment]
This looks to be an issue specifically with Next 13.5. Its compiling the plate/slate code differently, with 13.4 doing it correctly and 13.5 making it always error.

```
const hasInlines13_4 = (tN, h$) => h$.children.some((h$) => tlW.isText(h$) || tlO.isInline(tN, h$));

const hasInlines13_5 = function () {
  for (var uH, pZ = arguments.length, vB = Array(pZ), El = 0; El < pZ; El++) {
    vB[El] = arguments[El];
  }

  return (uH = lh), vB.children.some((lh) => td_.isText(lh) || tdc.isInline(uH, lh));
};
```

**Comment by KaneFreeman** [comment]
It appears to be a combination of slate `^0.94` (which Static CMS upgrades to in `v3.0.0`) and Next. Slate 0.94 changed how it handled editor functions (aka, what `hasInlines` is) and this result in a different output. This output is getting transformed by Next into non functioning code. This is likely related to [SWC minify](https://github.com/vercel/next.js/discussions/30237). I do not see an obvious answer to how to fix it, but I am still looking.

**Comment by KaneFreeman** [comment]
Looks like if `@staticcms/core` is not minified, then things work correctly. Next minifies it properly then. Since `@staticcms/core` is only meant for bundling, I set it to not be minified. `@staticcms/app` will continue to be minified.

This should be resolved with `v3.3.2`.
